### PR TITLE
RTL Support

### LINKED ISSUE #241: Text constraints issue in transactions list
## :writing_hand: Describe the bug
In case of long transaction url when `url` textview becomes 2 lines high it overlaps with `path` textview.
See attached screenshot (had to cover some text due to privacy reasons).

## :wrench: Expected behavior
Texts don't overlap

## :camera: Screenshots
texts_overlap

### PR DESCRIPTION
**Pull request by vbuberen**
## :camera: Screenshots

| Before | After |
| --- | --- |
| Before | After |
transactions_list_rtl_before | transactions_list_rtl

## :page_facing_up: Context
PR addresses feature request #117 
Since Chucker currently has min API 16 as supported, while RTL properties are available from 17, I added all RTL related things into styles in `values-v21`, which means that RTL works only from Android 5.0 (API 21) and newer.

## :pencil: Changes
- Added tags to define text direction and alignment in `styles.xml`
- Replaced `TableLayout` with `ConstrainLayout` in `TransactionOverviewFragment` layout.
- Added `useVectorDrawables = true` in `build.gradle` to avoid PNG generation during builds and use real vectors.
- Replaced `android:src` with `app:srcCompat` to use vector drawable via AndroidX library.
- Fixed constraints to close #241.

## :pencil: How to test
Set `Force RTL` in Developer options or switch to some RTL locale.

## :crystal_ball: Next steps
Would like to hear some feedback from real RTL users, who could tell if I did everything right. Since I am not an RTL user, can't claim that it works as it should.

### CODE REVIEW

**Comment by vbuberen** [comment]
This PR now closes #241 as well.

**Comment by vbuberen** [comment]
Let's check and merge it. It seems we won't get feedback from the original reporter of this feature request.
Also, the PR has some additional fixes, like fix #241 

Finally, due to #227 I updated ids for layout items to be 100% sure we are not clashing with ids in users projects.

File: library/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.chuckerteam.chucker">
-    <application>
+    <application
+      android:supportsRtl="true">
```
Comment (cortinico) [inline library/src/main/AndroidManifest.xml]: This should be avoided. Ideally we should not set a `android:supportsRtl` in the library as this gets merged in the final manifest. The default value for `supportsRtl` is false.

Moreover we might break builds for users that have `supportsRtl` set to `false` on the host app. See also https://github.com/ChuckerTeam/chucker/pull/121#issuecomment-533480441

File: library/src/main/res/values-v21/styles.xml [hunk 2]
```diff
@@ -1,5 +1,22 @@
 <?xml version="1.0" encoding="utf-8"?>
-<resources>
+<resources xmlns:tools="http://schemas.android.com/tools">
+
+   <style name="Chucker.Theme" parent="Theme.MaterialComponents.DayNight.NoActionBar">
```
Comment (cortinico) [inline library/src/main/res/values-v21/styles.xml]: Can I ask you to refactor to have a:

`<style name="Chucker.BaseTheme" parent="Theme.MaterialComponents.DayNight.NoActionBar">`

inside the `res/values/styles.xml` with all the common theme attributes?

We can then have the `Chucker.Theme` defined in both `values` and `values-v21` so we don't need to copy over the theme twice.

File: library/src/main/res/values-v21/styles.xml [hunk 3]
```diff
@@ -9,4 +26,15 @@
...
+    <style name="Chucker.TextAppearance.Value" parent="android:TextAppearance.Small">
+        <item name="android:paddingStart">16dp</item>
+        <item name="android:textIsSelectable">true</item>
+    </style>
+
+    <style name="Chucker.TextAppearance.TransactionTitle" parent="TextAppearance.AppCompat.Widget.ActionBar.Title">
+        <item name="android:gravity">top</item>
+        <item name="android:textSize" tools:ignore="SpUsage">16dp</item>
+        <item name="android:maxLines">2</item>
+    </style>
```
Comment (cortinico) [inline library/src/main/res/values-v21/styles.xml]: Why is this needed?

File: library/src/main/res/values-v21/styles.xml (hunk 2 above)
Comment (vbuberen) [reply library/src/main/res/values-v21/styles.xml]: Sure. I just thought that we might bump minSDK to 21 pretty soon due to Retrofit updates, etc.

File: library/src/main/res/values-v21/styles.xml (hunk 3 above)
Comment (vbuberen) [reply library/src/main/res/values-v21/styles.xml]: Just left it from original styles. It is used in `TransactionActivity` and `ErrorActivity` for toolbar title. Actually, I am against using dps for text sizes here. Will update as well.

File: library/src/main/res/values-v21/styles.xml (hunk 2 above)
Comment (cortinico) [reply library/src/main/res/values-v21/styles.xml]: Yeah true, but we don't have a PR pending for that (plus would be a major version bump). I think it's better if we keep the theme clean. Moreover having this setup with a `BaseTheme` could be handy for future similar scenarios

File: gradle.properties [hunk 4]
```diff
@@ -16,6 +16,8 @@ org.gradle.jvmargs=-Xmx1536m
 # http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
 org.gradle.parallel=true
 
+android.useAndroidX=true
```
Comment (vbuberen) [inline gradle.properties]: Added, because `3.6.0` plugin throws error without it.

**Comment by vbuberen** [comment]
@cortinico It is updated and ready for a new round of review.

File: library/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -8,9 +8,11 @@
             android:launchMode="singleTask"
             android:taskAffinity="com.chuckerteam.chucker.task"
             android:theme="@style/Chucker.Theme" />
+
         <activity android:name="com.chuckerteam.chucker.internal.ui.transaction.TransactionActivity"
             android:theme="@style/Chucker.Theme"
             android:parentActivityName="com.chuckerteam.chucker.internal.ui.MainActivity"/>
+
```
Comment (cortinico) [inline library/src/main/AndroidManifest.xml]: nit: whitespace change here that can be reverted

Comment (vbuberen) [reply library/src/main/AndroidManifest.xml]: I`ve added it for readability.

**Comment by vbuberen** [comment]
Just tested it myself - there is an issue due to `supportVectorDrawables = true` - Chucker crashes on devices with API 16-19. Will fix it later today. Do not merge, please.

**Comment by vbuberen** [comment]
Issues is fixed. In case someone interested in reasons here is the link to comment in the issue similar to what was seen: 
https://stackoverflow.com/a/38999661/7339798
